### ISSUE TITLE
🐛 Bug: Can break on certain non-ASCII cases

### ISSUE THREAD

**Issue by noinkling** [JoshuaKGoldberg/console-table-without-index#171]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/console-table-without-index/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

```js
console.table({'好好好好好': {a:'foo',b:'bar'}, '𝖮𝖧 𝖭𝖮': {a:'foo',b:'bar'}})
```
![image](https://github.com/user-attachments/assets/efbe2765-3581-4796-8613-3117ce67b083)

### Actual

```js
console.log(table({'好好好好好': {a:'foo',b:'bar'}, '𝖮𝖧 𝖭𝖮': {a:'foo',b:'bar'}}))
```
![image](https://github.com/user-attachments/assets/43f98e38-4083-4537-bcaa-e45d0e0abfcd)

### Additional Info

Some characters (like 好) display as double width despite being having `.length === 1`. Other characters have `.length === 2` because they're outside the unicode BMP (i.e. they're surrogate pairs), but display as single width.

`console.table()` accounts for this, but since you're using string `.slice()` based on an index obtained from the top row, it throws out the calculation.

You could change it to look for the second `│` on a row-by-row basis, however it would then break if that character was in the object key, so honestly I don't know how to fix it robustly without replicating [Node's internal width calculation](https://github.com/nodejs/node/blob/4e0f2773ff9465148a654dd0a2c1668104c20fec/lib/internal/util/inspect.js#L2333) (which as you can see delegates to ICU in the common case).

**Comment by JoshuaKGoldberg** [comment]
Yeah, this is tricky. I bet there's a package out there that already does the width calculation. https://www.npmjs.com/package/string-width, maybe?

Accepting PRs with the note that we should take on a dependency rather than reimplement all that logic in this package. Thanks!

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @noinkling for bug.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙
